pkg/storaged/btrfs/utils.jsx

```diff
@@ -88,3 +89,35 @@ export function validate_subvolume_name(name) {
     if (name.includes('/'))
         return cockpit.format(_("Name cannot contain the character '/'."));
 }
+
+// Determine whether BLOCK is a multi-device btrfs volume. This works
+// also when the volume is stored on LUKS devices, and some of them
+// are locked. (When all are locked, the volume does not exist at all
+// as far as Cockpit is concerned.)
+
+export function is_multi_device_btrfs_volume(block) {
+    // Check the direct case.
+    const block_btrfs = client.blocks_fsys_btrfs[block.path];
+    if (block_btrfs)
+        return block_btrfs.data.num_devices > 1;
+
+    // Might be a locked LUKS device. Try to figure out the uuid from
+    // its child fstab entries.
+    const block_crypto = client.blocks_crypto[block.path];
+    if (!block_crypto || client.blocks_cleartext[block.path])
+        return false;
+
+    for (const c of block_crypto.ChildConfiguration) {
+        if (c[0] == "fstab") {
+            const fsname = decode_filename(c[1].fsname.v);
+            const uuid_match = fsname.match(/^UUID=(?<uuid>[A-Fa-f0-9-]+)/);
+            if (uuid_match) {
+                const btrfs = client.uuids_btrfs_volume[uuid_match.groups.uuid];
+                if (btrfs && btrfs.data.num_devices > 1)
+                    return true;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20070-20240305-091506-a0863faa-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)